File: LeetCode/[E] - 933. Number of Recent Calls/solution.py
class RecentCounter:
    def __init__(self):
        self.queue = []
        self.head = 0

    def ping(self, t: int) -> int:
        self.queue.append(t)
        while self.queue[self.head] < t - 3000:
            self.head += 1
        return len(self.queue) - self.head



# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)

File: LeetCode/[E] - 933. Number of Recent Calls/solution.go
package main

type RecentCounter struct {
	queue []int
	head  int
}

func Constructor() RecentCounter {
	return RecentCounter{queue: []int{}}
}

func (rc *RecentCounter) Ping(t int) int {
	rc.queue = append(rc.queue, t)
	for rc.queue[rc.head] < t-3000 {
		rc.head++
	}
	return len(rc.queue) - rc.head
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Ping(t);
 */
